testing/conformance/benches/bench_utils.rs

```diff
@@ -0,0 +1,138 @@
+extern crate criterion;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::PathBuf;
+
+use conformance_tests::test_utils::*;
+use conformance_tests::vector::{ApplyMessage, MessageVector, Selector, TestVector, Variant};
+use conformance_tests::vm::{TestKernel, TestMachine};
+use criterion::*;
+use fvm::executor::{ApplyKind, DefaultExecutor, Executor};
+use fvm_shared::address::Protocol;
+use fvm_shared::blockstore::MemoryBlockstore;
+use fvm_shared::crypto::signature::SECP_SIG_LEN;
+use fvm_shared::encoding::Cbor;
+use fvm_shared::message::Message;
+
+pub fn apply_messages(
+    messages: &mut Vec<(Message, usize)>,
+    exec: &mut DefaultExecutor<TestKernel>,
+) {
+    // Apply all messages in the vector.
+    for (msg, raw_length) in messages.drain(..) {
+        // Execute the message.
+        // can assume this works because it passed a test before this ran
+        if let Err(e) = exec.execute_message(msg, ApplyKind::Explicit, raw_length) {
+            println!("fdsjfsdkjlafjksal; {:?}", e);
+            panic!();
+        };
+    }
+}
+
+pub fn bench_vector_variant(
+    group: &mut BenchmarkGroup<measurement::WallTime>,
+    name: String,
+    variant: &Variant,
+    v: &MessageVector,
+    bs: &MemoryBlockstore,
+) {
+    group.bench_function(name, move |b| {
+        b.iter_batched_ref(
+            || {
+                let v = v.clone();
+                let bs = bs.clone();
+                // TODO next few lines don't impact the benchmarks, but it might make them run waaaay more slowly... ought to make a base copy of the machine and exec and deepcopy them each time.
+                let machine = TestMachine::new_for_vector(v, variant, bs);
+                // can assume this works because it passed a test before this ran
+                machine.load_builtin_actors_modules().unwrap();
+                let exec: DefaultExecutor<TestKernel> = DefaultExecutor::new(machine);
+                let messages = v
+                    .apply_messages
+                    .iter()
+                    .map(|m| {
+                        let unmarshalled = Message::unmarshal_cbor(&m.bytes).unwrap();
+                        let mut raw_length = m.bytes.len();
+                        if unmarshalled.from.protocol() == Protocol::Secp256k1 {
+                            // 65 bytes signature + 1 byte type + 3 bytes for field info.
+                            raw_length += SECP_SIG_LEN + 4;
+                        }
+                        (unmarshalled, raw_length)
+                    })
+                    .collect();
+                (messages, exec)
+            },
+            |(messages, exec)| apply_messages(criterion::black_box(messages), exec),
+            BatchSize::LargeInput,
+        )
+    });
+}
+// how hard to do checks before running benchmark
+#[derive(Clone, Copy)]
+pub enum CheckStrength {
+    FullTest,
+    OnlyCheckSuccess,
+    NoChecks,
+}
+
+pub fn bench_vector_file(
```

Comment:
This function is too coarse. It is:
1. Maybe executing just one variant.
2. Maybe executing the messages in the vector, maybe not.
3. Overriding the name.

All depending on the input parameters. We should break this function up in smaller ones that can be composed depending on what the caller wants.

Also, the vector filtering (via the selector) should be done outside this method.